cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicController.java

```diff
@@ -0,0 +1,332 @@
+
+        super(namespace, cluster);
+        this.name = topicControllerName(cluster);
+        this.image = DEFAULT_IMAGE;
+        this.replicas = DEFAULT_REPLICAS;
+        this.healthCheckPath = "/";
+        this.healthCheckTimeout = DEFAULT_HEALTHCHECK_TIMEOUT;
+        this.healthCheckInitialDelay = DEFAULT_HEALTHCHECK_DELAY;
+
+        // create a default configuration
+        this.kafkaBootstrapServers = defaultBootstrapServers(cluster);
+        this.zookeeperConnect = defaultZookeeperConnect(cluster);
+        this.topicNamespace = namespace;
+        this.reconciliationInterval = DEFAULT_FULL_RECONCILIATION_INTERVAL;
+        this.zookeeperSessionTimeout = DEFAULT_ZOOKEEPER_SESSION_TIMEOUT;
+        this.topicConfigMapLabels = defaultTopicConfigMapLabels(cluster);
+    }
+
+    public void setTopicNamespace(String topicNamespace) {
+        this.topicNamespace = topicNamespace;
+    }
+
+    public String getTopicNamespace() {
+        return topicNamespace;
+    }
+
+    public void setTopicConfigMapLabels(String topicConfigMapLabels) {
+        this.topicConfigMapLabels = topicConfigMapLabels;
+    }
+
+    public String getTopicConfigMapLabels() {
+        return topicConfigMapLabels;
+    }
+
+    public void setReconciliationInterval(String reconciliationInterval) {
+        this.reconciliationInterval = reconciliationInterval;
+    }
+
+    public String getReconciliationInterval() {
+        return reconciliationInterval;
+    }
+
+    public void setZookeeperSessionTimeout(String zookeeperSessionTimeout) {
+        this.zookeeperSessionTimeout = zookeeperSessionTimeout;
+    }
+
+    public String getZookeeperSessionTimeout() {
+        return zookeeperSessionTimeout;
+    }
+
+    public void setKafkaBootstrapServers(String kafkaBootstrapServers) {
+        this.kafkaBootstrapServers = kafkaBootstrapServers;
+    }
+
+    public String getKafkaBootstrapServers() {
+        return kafkaBootstrapServers;
+    }
+
+    public void setZookeeperConnect(String zookeeperConnect) {
+        this.zookeeperConnect = zookeeperConnect;
+    }
+
+    public String getZookeeperConnect() {
+        return zookeeperConnect;
+    }
+
+    public static String topicControllerName(String cluster) {
+        return cluster + TopicController.NAME_SUFFIX;
+    }
+
+    protected static String defaultZookeeperConnect(String cluster) {
+        return ZookeeperCluster.zookeeperClusterName(cluster) + ":" + DEFAULT_ZOOKEEPER_PORT;
+    }
+
+    protected static String defaultBootstrapServers(String cluster) {
+        return KafkaCluster.kafkaClusterName(cluster) + ":" + DEFAULT_BOOTSTRAP_SERVERS_PORT;
+    }
+
+    protected static String defaultTopicConfigMapLabels(String cluster) {
+        String clusterLabel = String.format("%s=%s", ClusterController.STRIMZI_CLUSTER_LABEL, cluster);
+        String topicLabel = String.format("%s=%s", ClusterController.STRIMZI_KIND_LABEL, TopicController.KIND);
+        return clusterLabel + "," + topicLabel;
```

Comment:
Surely this would be better achieved using a single `return String.format()` with both labels, or using string concat. Mixing the two just seems inconsistent.

Reply:
Done